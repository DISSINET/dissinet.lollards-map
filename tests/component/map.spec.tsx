import "jest";
import * as React from "react";
import * as ReactDOM from "react-dom";

import { mount, shallow, render } from "enzyme";

import MapComponent from "./../../app/components/map";

describe("Map component", () => {
  beforeEach(async () => {});

  afterEach(async () => {});

  const mapMove = e => {};

  test("map component should load", async () => {
    const component = shallow(
      <MapComponent center={[10, 10]} zoom={7} handleMapMoved={mapMove} />
    );
    expect(component).toMatchSnapshot();
  });

  test("map component should have right center value", async () => {
    const component = shallow(
      <MapComponent center={[10, 10]} zoom={7} handleMapMoved={mapMove} />
    );
    const centerProp = component.find("Map").prop("center");
    expect(centerProp).toStrictEqual([10, 10]);
  });
});
