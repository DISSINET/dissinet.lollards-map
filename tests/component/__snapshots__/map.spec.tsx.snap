// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Map component map component should load 1`] = `
<div
  className="map"
  data-testid="map-wrapper"
>
  <Map
    center={
      Array [
        10,
        10,
      ]
    }
    onViewportChanged={[Function]}
    zoom={7}
  >
    <ForwardRef(Leaflet(LayersControl))
      position="topright"
    >
      <BaseLayer
        checked={true}
        name="OpenStreetMap.BlackAndWhite"
      >
        <ForwardRef(Leaflet(TileLayer))
          attribution="© <a href=\\"http://osm.org/copyright\\">OpenStreetMap</a> contributors"
          url="https://tiles.wmflabs.org/bw-mapnik/{z}/{x}/{y}.png"
        />
      </BaseLayer>
      <BaseLayer
        name="OpenStreetMap.Mapnik"
      >
        <ForwardRef(Leaflet(TileLayer))
          attribution="© <a href=\\"http://osm.org/copyright\\">OpenStreetMap</a> contributors"
          url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
        />
      </BaseLayer>
    </ForwardRef(Leaflet(LayersControl))>
    <ForwardRef(Leaflet(LayerGroup))>
      <ForwardRef(Leaflet(Marker))
        icon={
          NewClass {
            "_initHooksCalled": true,
            "options": Object {
              "iconAnchor": Array [
                12,
                40,
              ],
              "iconSize": Array [
                25,
                40,
              ],
              "iconUrl": "https://unpkg.com/leaflet@1.5.1/dist/images/marker-icon.png",
            },
          }
        }
        position={
          Array [
            48.93,
            18.15,
          ]
        }
      />
    </ForwardRef(Leaflet(LayerGroup))>
  </Map>
</div>
`;
